### PR TITLE
Improve UI

### LINKED ISSUE #2571: UI optimizations
**Is your feature request related to a problem? Please describe.**
A few issues:
1. Adding and deleting certain objects is cumbersome. For example, when deleting a question in a reader study, the user gets redirected to the overview page of the reader study. This is annoying when deleting several questions. This happens when deleting questions, readers, editors but also when adding readers and editors (also in other contexts like algorithms) 
2. Interfaces are not sorted or searchable in the drop down control when adding CIVs.
3. When setting the view content, there's no overview which interfaces are used in the reader study/algorithm/archive. It would be nice if this is displayed, or if there is an option to create a view content with all available interfaces, so the user does not have to copy or type them.

### PR DESCRIPTION
**Pull request by MikeOverkamp-diag**
Closes #2571

### CODE REVIEW

File: app/grandchallenge/components/views.py [hunk 1]
```diff
@@ -54,3 +55,24 @@ def get_context_data(self, *args, **kwargs):
...
         )
         return context
+
+
+class ComponentInterfaceAutocomplete(
+    LoginRequiredMixin, autocomplete.Select2QuerySetView
+):
+    def get_queryset(self):
+        qs = ComponentInterface.objects.filter(
+            kind__in=InterfaceKind.interface_type_image()
```
[the remark this answers is not in the document]
Comment (MikeOverkamp-diag) [reply app/grandchallenge/components/views.py:65]: I know, it needs to be updated for the new reader study forms. I'll create a new PR for that once https://github.com/comic/grand-challenge.org/pull/2575 has been merged, as to not create yet another PR into that one :slightly_smiling_face:

File: app/grandchallenge/components/views.py [hunk 2]
```diff
@@ -54,3 +55,24 @@ def get_context_data(self, *args, **kwargs):
             }
         )
         return context
+
+
+class ComponentInterfaceAutocomplete(
```
[the remark this answers is not in the document]
Comment (MikeOverkamp-diag) [reply app/grandchallenge/components/views.py:60]: So it is now used in all forms that use a normal select widget for the interface field that are on the main branch.

[the remark this answers is not in the document]
Comment (amickan) [reply app/grandchallenge/components/views.py:60]: Ah that makes sense! My bad for not looking close enough!

**Review by amickan — APPROVED** [review]

File: app/grandchallenge/components/views.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MikeOverkamp-diag) [reply app/grandchallenge/components/views.py:60]: No problem! I had to double check the algorithm form myself as well.

**Pull request merged**